gnd/src/main/java/com/google/android/gnd/ui/offlinebasemap/selector/OfflineBaseMapSelectorFragment.java

```diff
@@ -56,24 +57,15 @@ public static OfflineBaseMapSelectorFragment newInstance() {
   public void onCreate(@Nullable Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     viewModel = getViewModel(OfflineBaseMapSelectorViewModel.class);
-    // TODO: use the viewmodel
     Single<MapAdapter> mapAdapter = mapProvider.getMapAdapter();
     mapAdapter.as(autoDisposable(this)).subscribe(this::onMapReady);
-    viewModel.getDownloadMessages().observe(this, e -> e.ifUnhandled(this::onDownloadMessage));
-  }
-
-  private void onDownloadMessage(DownloadMessage message) {
-    switch (message) {
-      case STARTED:
-        popups.showSuccess(R.string.offline_base_map_download_started);
-        navigator.navigateUp();
-        break;
-      case FAILURE:
-      default:
-        popups.showError(R.string.offline_base_map_download_failed);
-        navigator.navigateUp();
-        break;
-    }
+    viewModel
+        .getDownloadMessages()
+        // Since we pop a toast, we need to observe on the main (UI) thread.
+        // Otherwise this subscription handler will trigger before Looper.prepare() has been called.
+        .observeOn(AndroidSchedulers.mainThread())
+        .as(autoDisposable(this))
+        .subscribe(this::onDownloadMessage);
```

Comment:
`LiveData` by default replays the last value on configuration changes. Since this isn't what we want in this case, we should use an `Observable` here instead.

Reply:
I'm missing something here, I think. The stream we're looking at `getDownloadMessages` is a `Single`, not LiveData.